### PR TITLE
Enable showing and fix existing warning

### PR DESCRIPTION
**Pull request by vvechkanov**
Enable warnings based on [this](https://msmobilecenter.visualstudio.com/Mobile-Center/_boards/board/t/IvanM-Team/Backlog%20Items/?workitem=79693) feature request.

### CODE REVIEW

File: Source/PLCrashAsyncThread_arm.c [hunk 1]
```diff
@@ -315,23 +315,23 @@ plcrash_greg_t plcrash_async_thread_state_get_reg (const plcrash_async_thread_st
...
-        case PLCRASH_ARM_R8: THREAD_STATE_SET(r[8], thread.ts_32, ts, regnum, reg); break;
-        case PLCRASH_ARM_R9: THREAD_STATE_SET(r[9], thread.ts_32, ts, regnum, reg); break;
-        case PLCRASH_ARM_R10: THREAD_STATE_SET(r[10], thread.ts_32, ts, regnum, reg); break;
-        case PLCRASH_ARM_R11: THREAD_STATE_SET(r[11], thread.ts_32, ts, regnum, reg); break;
-        case PLCRASH_ARM_R12: THREAD_STATE_SET(r[12], thread.ts_32, ts, regnum, reg); break;
-        case PLCRASH_ARM_SP: THREAD_STATE_SET(sp, thread.ts_32, ts, regnum, reg); break;
-        case PLCRASH_ARM_LR: THREAD_STATE_SET(lr, thread.ts_32, ts, regnum, reg); break;
-        case PLCRASH_ARM_PC: THREAD_STATE_SET(pc, thread.ts_32, ts, regnum, reg); break;
-        case PLCRASH_ARM_CPSR: THREAD_STATE_SET(cpsr, thread.ts_32, ts, regnum, reg); break;
+        case PLCRASH_ARM_R0: THREAD_STATE_SET(r[0], thread.ts_32, ts, regnum, (__uint32_t)reg); break;
```
Comment (MatkovIvan) [inline Source/PLCrashAsyncThread_arm.c:318]: Why not `uint32_t`?

File: Source/PLCrashFrameDWARFUnwind.cpp [hunk 2]
```diff
@@ -244,12 +244,12 @@ plframe_error_t plframe_cursor_read_dwarf_unwind (task_t task,
...
         PLCF_ASSERT(pc <= UINT64_MAX);
 
-        ferr = plframe_cursor_read_dwarf_unwind_int<uint64_t, int64_t>(task, pc, &image->macho_image, current_frame, previous_frame, next_frame);
+        ferr = plframe_cursor_read_dwarf_unwind_int<plcrash_greg_t, int64_t>(task, pc, &image->macho_image, current_frame, previous_frame, next_frame);
     } else {
         /* Could only happen due to programmer error; eg, an image that doesn't actually match our thread state */
         PLCF_ASSERT(pc <= UINT32_MAX);
 
-        ferr = plframe_cursor_read_dwarf_unwind_int<uint32_t, int32_t>(task, pc, &image->macho_image, current_frame, previous_frame, next_frame);
+        ferr = plframe_cursor_read_dwarf_unwind_int<plcrash_greg_t, int32_t>(task, pc, &image->macho_image, current_frame, previous_frame, next_frame);
```
Comment (MatkovIvan) [inline Source/PLCrashFrameDWARFUnwind.cpp:252]: It doesn't look like correct change

File: Source/PLCrashAsyncThread_arm.c (hunk 1 above)
Comment (vvechkanov) [reply Source/PLCrashAsyncThread_arm.c:318]: @olmisa

File: CrashReporter.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -3,7 +3,7 @@
 	archiveVersion = 1;
 	classes = {
 	};
-	objectVersion = 46;
+	objectVersion = 52;
```
Comment (MatkovIvan) [inline CrashReporter.xcodeproj/project.pbxproj:6]: This is breaking change. Revert

File: Source/PLCrashFrameDWARFUnwind.cpp [hunk 4]
```diff
@@ -244,12 +244,12 @@ plframe_error_t plframe_cursor_read_dwarf_unwind (task_t task,
...
         PLCF_ASSERT(pc <= UINT64_MAX);
 
-        ferr = plframe_cursor_read_dwarf_unwind_int<uint64_t, int64_t>(task, pc, &image->macho_image, current_frame, previous_frame, next_frame);
+        ferr = plframe_cursor_read_dwarf_unwind_int<int64_t, int64_t>(task, pc, &image->macho_image, current_frame, previous_frame, next_frame);
     } else {
         /* Could only happen due to programmer error; eg, an image that doesn't actually match our thread state */
         PLCF_ASSERT(pc <= UINT32_MAX);
 
-        ferr = plframe_cursor_read_dwarf_unwind_int<uint32_t, int32_t>(task, pc, &image->macho_image, current_frame, previous_frame, next_frame);
+        ferr = plframe_cursor_read_dwarf_unwind_int<int32_t, int32_t>(task, (int32_t)pc, &image->macho_image, current_frame, previous_frame, next_frame);
```
Comment (MatkovIvan) [inline Source/PLCrashFrameDWARFUnwind.cpp:252]: Revert the type

File: Source/PLCrashAsyncThread_arm.c (hunk 1 above)
Comment (olmisa) [reply Source/PLCrashAsyncThread_arm.c]: fixed

File: Source/PLCrashAsyncDwarfCFAStateEvaluation.cpp [hunk 5]
```diff
@@ -227,11 +227,11 @@ plcrash_error_t dwarf_cfa_state<machine_ptr, machine_ptr_s>::eval_program (plcra
                 break;
                 
             case DW_CFA_def_cfa:
-                set_cfa_register(dw_expr_read_uleb128_regnum(), dw_expr_read_uleb128());
+                set_cfa_register(dw_expr_read_uleb128_regnum(), (machine_ptr_s)dw_expr_read_uleb128());
```
Comment (landonf) [inline Source/PLCrashAsyncDwarfCFAStateEvaluation.cpp:230]: Here (and elsewhere below), shouldn't `machine_ptr` should be used when operating on unsigned DWARF CFA register values?

Comment (olmisa) [reply Source/PLCrashAsyncDwarfCFAStateEvaluation.cpp]: Fixed

**Comment by MatkovIvan** [comment]
CI highlights a real issue. @vvechkanov @olmisa please fix the tests

**Comment by olmisa** [comment]
@russelarms warnings in tests were fixed.

**Pull request merged**